inference-engine/src/vpu/graph_transformer/src/frontend/detect_network_batch.cpp

```diff
@@ -83,93 +73,80 @@ void FrontEnd::detectNetworkBatch(
         auto ieShapes = ieData->getTensorDesc().getDims();
         env.log->trace("Input [%s] : %v", p.first, ieShapes);
 
-        switch (ieData->getLayout()) {
-        case ie::Layout::NCDHW:
-        case ie::Layout::NDHWC:
-        case ie::Layout::NCHW:
-        case ie::Layout::NHWC:
-        case ie::Layout::NC:
-        case ie::Layout::CN:
+        if (ieShapes.size() == 4 || ieShapes.size() == 5) {
+            auto batch = ieShapes[0];
+            if (!inputBatch)
+                inputBatch = batch;
+
+            if (inputBatch != batch) {
+                env.log->trace("Network has inputs with different batch dim, aborting batch detection");
+                return;
+            }
+
             ieShapes[0] = 1;
-            break;
-        default:
-            VPU_THROW_FORMAT("Input %v has unexpected layout %v", ieData->getName(), ieData->getLayout());
+        } else {
+            // assume only 4D and 5D inputs have batch
```

Comment:
Nitpick: personally I don't see a reason for such commentaries, code explains it better